dat/dat.go

```diff
@@ -63,9 +71,15 @@ func Parse(dat []byte) Dat {
 }
 
 // FindByCRC loops over the Dats in the DB and concurrently matches CRC checksums.
-func (db *DB) FindByCRC(romPath string, romName string, crc uint32, games chan (Game)) {
+func (db *DB) FindByCRC(romPath string, romName string, crc uint32, games chan (Game)) (bool) {
 	var wg sync.WaitGroup
 	wg.Add(len(*db))
+	// this structure and subsequent object are remade ever run.
+	type SafeBool struct {
+		mu    sync.Mutex
+		found bool
+	}
+	game_found := SafeBool{found: false}
```

Comment:
Didn't analyze it deeply, but do you really need a mutex you a `atomic.Bool` fits your needs?